src/QueryPipeline/QueryPipelineBuilder.cpp

```diff
@@ -323,7 +324,15 @@ std::unique_ptr<QueryPipelineBuilder> QueryPipelineBuilder::joinPipelines(
     right->pipe.dropExtremes();
 
     left->pipe.collected_processors = collected_processors;
-    right->pipe.collected_processors = collected_processors;
+    IQueryPlanStep* step = right->pipe.processors.back()->getQueryPlanStep();
+    if (ExpressionStep::name == step->getName())
+    {
+        right->pipe.collected_processors = static_cast<ExpressionStep*>(step)->getProcessors();
```

Comment:
isn't `right->pipe.collected_processors` equal to `right->pipe.processors.back()->getQueryPlanStep()->getProcessors()`?

Reply:
No. right->pipe.collected_processors is nullptr before assignment.